pandas/core/arrays/arrow/array.py

```diff
@@ -628,7 +629,9 @@ def __setstate__(self, state) -> None:
 
     def _cmp_method(self, other, op):
         pc_func = ARROW_CMP_FUNCS[op.__name__]
-        if isinstance(other, (ArrowExtensionArray, np.ndarray, list, BaseMaskedArray)):
+        if isinstance(
+            other, (ArrowExtensionArray, np.ndarray, list, BaseMaskedArray)
+        ) or isinstance(getattr(other, "dtype", None), CategoricalDtype):
```

Comment:
So for Categorical, we don't need to extract the underlying values?

Reply:
That's what _box_pa is doing, isn't it?

Reply:
True, I guess in _box_pa `pa.array` is correctly handling it luckily which is good.